### ISSUE TITLE
some x64dbg plugin failed because of the new implements of pluginexprfuncregister function

### ISSUE THREAD

**Issue by hackoflpf** [x64dbg/x64dbg#2737]
### Operating System

win10

### x64dbg Version

jul 1 2021

### Describe the issue

when i use the plugin strmatch ,i find that it can only success for one time,then it won't work as expected.I modify the source code of strmatch plugin and print some logs ,i find that every time the argv passed by program is the same as first time.so i checked the source code and i found the 3ab89502c commit has modifed the plugin_loader.cpp which cause this big problem.I think the developers should review the code because the original code works well,the new code use vector to store a argv,but it use push_back to store it ,so every time the plugin will get the first argv....

```
+struct ExprFuncWrapper
+{
+    void* user;
+    int argc;
+    CBPLUGINEXPRFUNCTION cbFunc;
+    std::vector<duint> cbArgv;
+
+    static bool callback(ExpressionValue* result, int argc, const ExpressionValue* argv, void* userdata)
+    {
+        auto cbUser = reinterpret_cast<ExprFuncWrapper*>(userdata);
+        for(auto i = 0; i < argc; i++)
+            cbUser->cbArgv.push_back(argv[i].number);
+
+        result->type = ValueTypeNumber;
+        result->number = cbUser->cbFunc(argc, cbUser->cbArgv.data(), cbUser->user);
+
+        return true;
+    }
+};
+
```

### Steps to reproduce

1. use strmatch plugin command strmatch_set to set a index
2. use strcmp_local() to compare a string when hit a breakpoint 
3. it will success for only one time.

**Comment by mrexodia** [comment]
It’s now implemented inside of x64dbg: https://github.com/x64dbg/x64dbg/blob/development/src/dbg/expressionfunctions.cpp#L151

**Comment by torusrxxx** [comment]
shouldn't be an excuse for not supporting strmatch

**Comment by mrexodia** [comment]
Was fixed by 31bec6102a9f64b0203542886d509e611e9fffa7
